### PR TITLE
Making primitive cooking real part 1 - Baking bread and 'microwaving' rice in furnaces

### PR DESCRIPTION
**Pull request by Paxilmaniac**
## About The Pull Request
The first pr of a few I hope to complete sometime soon that makes primitive cooking possible without the use of machines.

This first one allows both microwave and oven cooking actions to be completed at a forge with heat. How do you do this, you might ask? It's easy, grab a food item and use it on a forge, you'll be a given a prompt as to if you want to 'microwave' or oven cook the item. Choose one and after a few seconds... delicious baked goods! Or... a burned mess if the item you tried cooking isn't meant to be cooked that way.

## How This Contributes To The Skyrat Roleplay Experience
I always thought it was kinda silly that we had to staple on a kitchen with power to the ashwalker camp in order for them to cook most of our foods, so I'm gonna work towards fixing that.

image

We'll get there someday

## Changelog

:cl:
qol: Food can now be baked and 'microwaved' at lit forges, removing the need for an oven and microwave at the ashwalker camp completely
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/reagent_forging/code/forge.dm [hunk 1]
```diff
@@ -462,6 +467,40 @@
...
+	if(istype(attacking_item, /obj/item/food))
+		var/obj/item/food/thing_to_cook = attacking_item
+		if(in_use)
+			to_chat(user, span_warning("You cannot do multiple things at the same time!"))
+			return
+		in_use = TRUE
+		if(forge_temperature < MIN_FORGE_TEMP)
+			fail_message(user, "The [src] is not hot enough to start cooking [thing_to_cook]!")
+			return
+		var/datum/user_input = tgui_input_list(user, "Select a machine to emulate the cooking effects of.", "Cooking Selection", cooking_choice)
```
Comment (tf-4) [inline modular_skyrat/modules/reagent_forging/code/forge.dm]: Honestly, I think a radial menu would be nicer here. See how the reagent grinder does it. Would need some new icons, but they can just be smaller versions of the oven and microwave respectively.
Something like;
```suggestion
	    var/static/radial_oven = image(icon = 'modular_skyrat/modules/reagent_forging/icons/radial.dmi, icon_state = "radial_oven")
	    var/static/radial_microwave = image(icon = 'modular_skyrat/modules/reagent_forging/icons/radial.dmi', icon_state = "radial_microwave")
        var/list/options = list()
        options["oven"] = radial_oven
        options["microwave"] = radial_microwave
        if(length(options) < 1)
		    return
	    if(length(options) == 1)
		    for(var/key in options)
			    choice = key
	    else
		    choice = show_radial_menu(user, src, options)
```
(untested, donut commit)

Comment (Paxilmaniac) [reply modular_skyrat/modules/reagent_forging/code/forge.dm]: I had thought of making it a radial menu, just didn't make one out of not knowing how they work, let me see what I can do

**Comment by twilightwanderer** [comment]
Surely such mechanics could be added to the upstream. It seems logical enough.

**Comment by jjpark-kb** [comment]
> Surely such mechanics could be added to the upstream. It seems logical enough.

what? upstream doesn't have forging.

File: modular_skyrat/modules/reagent_forging/code/forge.dm [hunk 2]
```diff
@@ -68,10 +68,16 @@
 		"Shovel" = /obj/item/forging/incomplete/shovel,
 		"Arrowhead" = /obj/item/forging/incomplete/arrowhead,
 	)
+	///radial buttons for cooking food
+	var/static/radial_oven = image(icon = 'modular_skyrat/modules/reagent_forging/icons/hud/forge_radials.dmi', icon_state = "oven")
+	var/static/radial_microwave = image(icon = 'modular_skyrat/modules/reagent_forging/icons/hud/forge_radials.dmi', icon_state = "microwave")
+
+	var/static/list/radial_options = list("oven" = radial_oven, "microwave" = radial_microwave)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/reagent_forging/code/forge.dm]: Document each of these individually, please!

File: modular_skyrat/modules/reagent_forging/code/forge.dm [hunk 3]
```diff
@@ -68,10 +68,16 @@
...
 
 /obj/structure/reagent_forge/examine(mob/user)
 	. = ..()
 	. += span_warning("<br>Perhaps using your hand on [src] when skilled will do something...<br>")
+	. += span_notice("You could probably use [src] to bake and cook things...<br>")
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/reagent_forging/code/forge.dm]:
```suggestion
	. += span_notice("You could probably use [src] to <b>bake</b> or <b>cook</b> food...<br>")
```

**Review by Zonespace27 — APPROVED** [review]

**Review by tf-4 — APPROVED** [review]

File: modular_skyrat/modules/reagent_forging/code/forge.dm [hunk 4]
```diff
@@ -462,6 +468,41 @@
...
+		if(!do_after(user, 10 SECONDS, target = src))
+			fail_message(user, "You stop trying to cook [thing_to_cook]!")
+			return
+		switch(user_input)
+			if("oven")
+				var/datum/component/bakeable/item_bakeable_component = thing_to_cook.GetComponent(/datum/component/bakeable)
+				if(item_bakeable_component.bake_result)
+					item_to_spawn = item_bakeable_component.bake_result
+				else
+					item_to_spawn = /obj/item/food/badrecipe
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/reagent_forging/code/forge.dm]: You could probably do this with a ternary operator instead (`var/my_var = a == 1 ? value_if_true : value_if_false`)

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/reagent_forging/code/forge.dm (hunk 4 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/reagent_forging/code/forge.dm]: these scare me

File: modular_skyrat/modules/reagent_forging/code/forge.dm [hunk 5]
```diff
@@ -462,6 +470,35 @@
...
+		switch(user_input)
+			if("oven")
+				var/datum/component/bakeable/item_bakeable_component = thing_to_cook.GetComponent(/datum/component/bakeable)
+				item_to_spawn = item_bakeable_component.bake_result ? item_bakeable_component.bake_result : /obj/item/food/badrecipe
+			if("microwave")
+				item_to_spawn = thing_to_cook.microwaved_type ? thing_to_cook.microwaved_type : /obj/item/food/badrecipe
+		qdel(thing_to_cook)
+		new item_to_spawn(get_turf(src))
+		in_use = FALSE
+		return TRUE
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/reagent_forging/code/forge.dm]: First of all, if you're going to have a proc that's meant to return true at the end, then you're going to want to explicitly return false when it fails. I know the rest of the proc wasn't doing it, but I also forgot to mention that back when that code was first added.

Secondly, let it breathe, space shit out, it really helps readability. We're not running out of space for whitespaces, so don't be shy and use 'em.
```suggestion
	if(istype(attacking_item, /obj/item/food))
		var/obj/item/food/thing_to_cook = attacking_item

		if(in_use)
			to_chat(user, span_warning("You cannot do multiple things at the same time!"))
			return FALSE

		in_use = TRUE
	
		if(forge_temperature < MIN_FORGE_TEMP)
			fail_message(user, "The [src] is not hot enough to start cooking [thing_to_cook]!")
			return FALSE

		var/user_input = show_radial_menu(user, src, radial_options)
		var/obj/item_to_spawn

		if(!user_input)
			fail_message(user, "No choice made")
			return FALSE

		balloon_alert_to_viewers("cooking...")

		if(!do_after(user, 10 SECONDS, target = src))
			fail_message(user, "You stop trying to cook [thing_to_cook]!")
			return FALSE

		switch(user_input)
			if("oven")
				var/datum/component/bakeable/item_bakeable_component = thing_to_cook.GetComponent(/datum/component/bakeable)
				item_to_spawn = item_bakeable_component.bake_result ? item_bakeable_component.bake_result : /obj/item/food/badrecipe
			if("microwave")
				item_to_spawn = thing_to_cook.microwaved_type ? thing_to_cook.microwaved_type : /obj/item/food/badrecipe

		qdel(thing_to_cook)
		new item_to_spawn(get_turf(src))
		in_use = FALSE
		return TRUE
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/reagent_forging/code/forge.dm (hunk 5 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/reagent_forging/code/forge.dm]: should go through when I have time and update the rest of this to meet that then, if someone doesn't get it first

**Comment by Paxilmaniac** [comment]
@GoldenAlpharex GOOOOOOOOOOOOLDEN

**Review by GoldenAlpharex — APPROVED** [review]
I forgor 💀

**Pull request merged**